### PR TITLE
[SW-1504] Apply type converters to rest of the PySparkling

### PR DESCRIPTION
**Pull request by jakubhava**

### CODE REVIEW

File: py/ai/h2o/sparkling/ml/algos/H2ODeepLearning.py [hunk 1]
```diff
@@ -16,53 +16,18 @@
...
-class H2ODeepLearning(H2ODeepLearningParams, JavaEstimator, JavaMLReadable, JavaMLWritable):
+class H2ODeepLearning(H2ODeepLearningParams, H2OAlgoBase):
 
     @keyword_only
-    def __init__(self, modelId=None, splitRatio=1.0, labelCol="label", weightCol=None, featuresCols=[], allStringColumnsToCategorical=True, columnsToCategorical=[],
-                 nfolds=0, keepCrossValidationPredictions=False, keepCrossValidationFoldAssignment=False, parallelizeCrossValidation=True,
-                 seed=-1, distribution="AUTO", epochs=10.0, l1=0.0, l2=0.0, hidden=[200,200], reproducible=False,
-                 convertUnknownCategoricalLevelsToNa=False, foldCol=None, predictionCol="prediction", detailedPredictionCol="detailed_prediction",
-                 withDetailedPredictionCol=False, convertInvalidNumbersToNa=False, **deprecatedArgs):
+    def __init__(self, **kwargs):
```
Comment (mn-mikke) [inline py/ai/h2o/sparkling/ml/algos/H2ODeepLearning.py]: I really like the compactness of the code, but wouldn't users loss an opportunity to see IDE hints with the list of possible constructor arguments?

File: py/ai/h2o/sparkling/ml/algos/H2OXGBoost.py [hunk 2]
```diff
@@ -16,84 +16,18 @@
...
-                                                                                       "sampleRate", "subsample", "colSampleRate", "colSampleByLevel", "colSampleRatePerTree",
-                        "colsampleBytree", "maxAbsLeafnodePred", "maxDeltaStep", "minSplitImprovement", "gamma",
-                        "minSumHessianInLeaf", "minDataInLeaf", "regLambda", "regAlpha", "rateDrop", "skipDrop"]
-        set_double_values(kwargs, double_types)
-        return self._set(**kwargs)
-
-    def _create_model(self, java_model):
-        return H2OMOJOModel(java_model)
+        self._setDefaultValuesFromJava()
+        self._set(**kwargs)
```
Comment (mn-mikke) [inline py/ai/h2o/sparkling/ml/algos/H2OXGBoost.py]: nit: missing new line

File: py/ai/h2o/sparkling/ml/params/H2OGBMParams.py [hunk 3]
```diff
@@ -19,18 +19,24 @@
...
     ##
     # Param definitions
     ##
-    learnRate = Param(Params._dummy(), "learnRate", "Learning rate (from 0.0 to 1.0)")
-    learnRateAnnealing = Param(Params._dummy(), "learnRateAnnealing", "Scale the learning rate by this factor after each tree (e.g., 0.99 or 0.999)")
-    colSampleRate = Param(Params._dummy(), "colSampleRate", "Column sample rate (from 0.0 to 1.0)")
-    maxAbsLeafnodePred = Param(Params._dummy(), "maxAbsLeafnodePred", "Maximum absolute value of a leaf node prediction")
-    predNoiseBandwidth = Param(Params._dummy(), "predNoiseBandwidth", "Bandwidth (sigma) of Gaussian multiplicative noise ~N(1,sigma) for tree node predictions")
+    learnRate = Param(Params._dummy(), "learnRate", "Learning rate (from 0.0 to 1.0)",
+                      H2OTypeConverters.toFloat())
```
Comment (mn-mikke) [inline py/ai/h2o/sparkling/ml/params/H2OGBMParams.py]: WDYT about checking also ranges of float parameters?

File: py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py [hunk 4]
```diff
@@ -15,58 +15,57 @@
...
-    rateDrop = Param(Params._dummy(), "rateDrop", "rate drop")
-    oneDrop = Param(Params._dummy(), "oneDrop", "onde drop")
-    skipDrop = Param(Params._dummy(), "skipDrop", "skip drop")
-    gpuId = Param(Params._dummy(), "gpuId", "GPU id")
-    backend = Param(Params._dummy(), "backend", "Backend")
+    quietMode = Param(Params._dummy(), "quietMode", "Quiet mode", H2OTypeConverters.toBoolean())
+    ntrees = Param(Params._dummy(), "ntrees", "Number of trees", H2OTypeConverters.toInt())
+    nEstimators = Param(Params._dummy(), "nEstimators", "number of estimators", H2OTypeConverters.toInt())
+    maxDepth = Param(Params._dummy(), "maxDepth", "Maximal depth", H2OTypeConverters.toInt())
+    minRows = Param(Params._dummy(), "minRows", "Min rows", H2OTypeConverters.toFloat())
```
Comment (mn-mikke) [inline py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py]: Any idea why this could be a float number? I would expect integers only...

File: py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py [hunk 5]
```diff
@@ -15,58 +15,57 @@
...
+    minChildWeight = Param(Params._dummy(), "minChildWeight", "minimal child weight", H2OTypeConverters.toFloat())
+    learnRate = Param(Params._dummy(), "learnRate", "learn rate", H2OTypeConverters.toFloat())
+    eta = Param(Params._dummy(), "eta", "eta", H2OTypeConverters.toFloat())
+    learnRateAnnealing = Param(Params._dummy(), "learnRateAnnealing", "Learn Rate Annealing", H2OTypeConverters.toFloat())
+    sampleRate = Param(Params._dummy(), "sampleRate", "Sample rate", H2OTypeConverters.toFloat())
+    subsample = Param(Params._dummy(), "subsample", "subsample", H2OTypeConverters.toFloat())
+    colSampleRate = Param(Params._dummy(), "colSampleRate", "col sample rate", H2OTypeConverters.toFloat())
+    colSampleByLevel = Param(Params._dummy(), "colSampleByLevel", "Col Sample By Level", H2OTypeConverters.toFloat())
+    colSampleRatePerTree = Param(Params._dummy(), "colSampleRatePerTree", "col samle rate", H2OTypeConverters.toFloat())
```
Comment (mn-mikke) [inline py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py]: I now that this is irrelevant to this change, jus noticed that the description is the same as for `colSampleRate`.

File: py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py [hunk 6]
```diff
@@ -15,58 +15,57 @@
...
+    colsampleBytree = Param(Params._dummy(), "colsampleBytree", "col sample by tree", H2OTypeConverters.toFloat())
```
Comment (mn-mikke) [inline py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py]: It seems that this parameter deserves to be renamed in a different PR.

File: py/ai/h2o/sparkling/ml/algos/H2ODeepLearning.py (hunk 1 above)
Comment (jakubhava) [reply py/ai/h2o/sparkling/ml/algos/H2ODeepLearning.py]: Yup, will keep the params in init method

File: py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py (hunk 4 above)
Comment (jakubhava) [reply py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py]: Yup, it is weird, but as H2O accepts Double values wanted to be consistent

File: py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py (hunk 5 above)
Comment (jakubhava) [reply py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py]: Great catch, will fix in subsequent PR

File: py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py (hunk 6 above)
Comment (jakubhava) [reply py/ai/h2o/sparkling/ml/params/H2OXGBoostParams.py]: Great catch, will fix in subsequent PR

File: py/ai/h2o/sparkling/ml/params/H2OGBMParams.py (hunk 3 above)
Comment (jakubhava) [reply py/ai/h2o/sparkling/ml/params/H2OGBMParams.py]: Perfect idea -> created jira https://0xdata.atlassian.net/browse/SW-1507
